memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionRequest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.memory.action.conversation;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.INTERACTIONS_ADDITIONAL_INFO_FIELD;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+public class UpdateInteractionRequest extends ActionRequest {
+    private String interactionId;
+    private Map<String, Object> updateContent;
+
+    private static final Set<String> allowedList = new HashSet<>(Arrays.asList(INTERACTIONS_ADDITIONAL_INFO_FIELD));
+
+    @Builder
+    public UpdateInteractionRequest(String interactionId, Map<String, Object> updateContent) {
+        this.interactionId = interactionId;
+        this.updateContent = filterUpdateContent(updateContent);
+    }
+
+    public UpdateInteractionRequest(StreamInput in) throws IOException {
+        super(in);
+        this.interactionId = in.readString();
+        this.updateContent = filterUpdateContent(in.readMap());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(this.interactionId);
+        out.writeMap(this.getUpdateContent());
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException exception = null;
+
+        if (this.interactionId == null) {
+            exception = addValidationError("interaction id can't be null", exception);
+        }
+        if (this.updateContent == null) {
+            exception = addValidationError("Update Interaction content can't be null", exception);
+        }
+
+        return exception;
+    }
+
+    public static UpdateInteractionRequest parse(XContentParser parser, String interactionId) throws IOException {
+        Map<String, Object> dataAsMap = null;
+        dataAsMap = parser.map();
+
+        if (dataAsMap == null) {
+            throw new OpenSearchParseException("Failed to parse UpdateInteractionRequest due to Null update content");
```

Comment:
Can we add a test?

Reply:
@Test
    public void testConstructor_NullUpdateContent() throws IOException {

This test is added. Probably not exactly what you wanted :). The constructor does not allow this dataAsMap to null. We can add another tests to construct a null updateContent in the next PR.

Reply:
No, I'm not asking for that test. I am asking to extend this test `testParse_Success` where we provide a parser which could be null, after parsing (`parser.map()`)

Reply:
> provide a parser which could be null

@dhrubo-os , you mean test `parser` is null case ?
@Zhangxunmt Is it possible that `parser` will be null?

Reply:
Yeah.

Reply:
This parser only comes from the RestAction. In the Rest layer, the RestRequest is guaranteed not be Null, otherwise exception is throw before reaching this point. We can manually create a request with Null parser, but this would only exists in the test.

Reply:
[nit] In summary,  after `dataAsMap = parser.map();`, `dataAsMap` shouldn't be null. May be we can simplify the code by removing that unnecessary null check.